Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -1116,6 +1163,7 @@ public void Show()
 
         public async void Hide()
         {
+            lastHistoryIndex = 1;
```

Comment:
Why is this reset needed?

Reply:
> If you close the window and start the cycle again, you should see the most recent query first.

If a user set last query style to preserve or selected that means he wants to continue the working state if so. Maybe better to reset it only when empty?

Reply:
> Why is this reset needed?

When the user closes the window, the index is initialized. Lastemptyquery is just a setting about query input, so it shouldn't have anything to do with this setting. It's kind of weird when I try it in  without doing this initialization. Get rid of it and try it. It's weird.

Reply:
> > If you close the window and start the cycle again, you should see the most recent query first.
> 
> If a user set last query style to preserve or selected that means he wants to continue the working state if so. Maybe better to reset it only when empty?

@VictoriousRaptor When the user closes the window, the index is initialized. Lastemptyquery is just a setting about query input, so it shouldn't have anything to do with this setting. It's kind of weird when I try it in  without doing this initialization. Get rid of it and try it. It's weird.

It is also not a good idea to initialize when user delete a query. Not closing the window means that the user intends to keep it that way. When using the function in other cli programs, the index is maintained even if the query is deleted. This is a feature that cli users will mainly use, so it's right to follow this UX.